pkg/manifests/manifests.go

```diff
@@ -2140,6 +2140,20 @@ func (f *Factory) MetricsServerDeployment() (*appsv1.Deployment, error) {
 	containers[idx].Args = f.setTLSSecurityConfiguration(podSpec.Containers[0].Args,
 		MetricsServerTLSCipherSuitesFlag, MetricsServerTLSMinTLSVersionFlag)
 
+	// Hash the Kubelet Serving CA Bundle configmap value and propagate it as a label to the
```

Comment:
I'm afraid the values will be overridden and only `hashByteMap(metricsClientCert.Data)` will be applied.
we should either:
- Group the 3 hashes and put them under the same label, I don't think `monitoring.openshift.io/hash` is the suitable name as it may mean the hash of the resource itself (Pod in this case) which may be confusing. Maybe we could use something like `monitoring.openshift.io/datastore-hash` or something.
- have 3 different labels `monitoring.openshift.io/CONFIGMAP-OR-SECRET-NAME-hash` for each configmap/secret.

WDYT?

If we move away from `monitoring.openshift.io/hash`, we could use annotations, in the Helm world, we use sth like that, they're called checksums https://github.com/search?q=repo%3Abitnami%2Fcharts%20checksum&type=code maybe we could use that term instead of hash to align with Helm.

Reply:
Thats a good catch, I think we should have separate label/annotation in that case. 
For telemeter also we use annotation https://github.com/openshift/cluster-monitoring-operator/blob/70998442501d8a2d0d35dbb493a623161b3c56a3/pkg/manifests/manifests.go#L3121